components/feature/pwa/src/test/java/mozilla/components/feature/pwa/ManifestStorageTest.kt

```diff
@@ -0,0 +1,94 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+package mozilla.components.feature.pwa
+
+import android.util.AtomicFile
+import androidx.test.ext.junit.runners.AndroidJUnit4
+import kotlinx.coroutines.ExperimentalCoroutinesApi
+import kotlinx.coroutines.runBlocking
+import mozilla.components.concept.engine.manifest.WebAppManifest
+import mozilla.components.concept.engine.manifest.WebAppManifestParser
+import mozilla.components.support.test.mock
+import mozilla.components.support.test.robolectric.testContext
+import org.junit.Assert.assertEquals
+import org.junit.Assert.assertNull
+import org.junit.Test
+import org.junit.runner.RunWith
+import org.mockito.Mockito.`when`
+import org.mockito.Mockito.doReturn
+import org.mockito.Mockito.spy
+import org.mockito.Mockito.verify
+import java.io.FileOutputStream
+
+@ExperimentalCoroutinesApi
+@RunWith(AndroidJUnit4::class)
+class ManifestStorageTest {
+    @Test
+    fun `load returns null if entry does not exist or is invalid`() = runBlocking {
+        val storage = spy(ManifestStorage(testContext))
+        assertNull(storage.loadManifest("https://example.com"))
+
+        val mozFile = storage.mockFile("https://mozilla.org")
+        val fireFile = storage.mockFile("https://firefox.com")
+        `when`(mozFile.readFully()).thenReturn("[]".toByteArray())
```

Comment:
nit: What do you think about using `whenever()` here? Backtick-less notation is bit more clear as for me.